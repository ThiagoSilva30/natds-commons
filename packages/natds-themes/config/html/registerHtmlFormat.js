import fs from 'fs';
import path from 'path';
import Handlebars from 'handlebars';
import variables from '../nodeVariables.cjs';

const { __dirname } = variables;

const registerHtmlFormat = () => {
  const navigationTemplate = fs.readFileSync(path.resolve(__dirname, './html/templates/navigation.hbs')).toString();

  Handlebars.registerPartial('navigation', navigationTemplate);

  const html = fs.readFileSync(path.resolve(__dirname, './html/templates/colors.hbs')).toString();
  const htmlTemplate = Handlebars.compile(html);

  const formatter = dictionary => htmlTemplate({
    color: dictionary.properties.color,
  });

  return {
    formatter,
    name: 'html/colors',
  };
};

export default registerHtmlFormat;
