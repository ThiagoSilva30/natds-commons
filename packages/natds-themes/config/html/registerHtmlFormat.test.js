import registerHtmlFormat from './registerHtmlFormat';

const dictionary = {
  properties: {
    color: {
      primary: {
        attributes: {},
        name: 'colorPrimary',
        original: {
          value: '#F091C9',
        },
        path: [
          'color',
          'primary',
        ],
        value: '#F091C9',
      },
    },
  },
};

describe('registerHtmlFormat', () => {
  it('should return the register config', () => {
    const register = registerHtmlFormat();

    expect(typeof register.formatter).toBe('function');
    expect(register.name).toBe('html/colors');
  });

  it('should return the correct formatter template', () => {
    const register = registerHtmlFormat();

    expect(register.formatter(dictionary)).toMatchSnapshot();
  });
});
