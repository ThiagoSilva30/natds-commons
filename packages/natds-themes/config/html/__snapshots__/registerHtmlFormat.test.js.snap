// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registerHtmlFormat should return the correct formatter template 1`] = `
"<html lang=\\"en\\">

<head>
  <meta charset=\\"UTF-8\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
  <title>Color tokens</title>
  <link rel=\\"stylesheet\\" href=\\"https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/css/bootstrap.min.css\\"
    integrity=\\"sha384-9aIt2nRpC12Uk9gS9baDl411NQApFmC26EwAOH8WgZl5MYYxFfc+NcPb1dKGj7Sk\\" crossorigin=\\"anonymous\\">
</head>

<body>
  <style>
    .color-sample {
      border: 1px solid #eaeaea;
      height: 50px;
    }
  </style>
  <nav class=\\"navbar navbar-expand-lg navbar-light bg-light\\">
    <button class=\\"navbar-toggler\\" type=\\"button\\" data-toggle=\\"collapse\\" data-target=\\"#navbarSupportedContent\\"
      aria-controls=\\"navbarSupportedContent\\" aria-expanded=\\"false\\" aria-label=\\"Toggle navigation\\">
      <span class=\\"navbar-toggler-icon\\"></span>
    </button>
  
    <div class=\\"collapse navbar-collapse\\" id=\\"navbarSupportedContent\\">
      <ul id=\\"wrapper\\" class=\\"navbar-nav mr-auto\\">
  
      </ul>
    </div>
  </nav>
  
  <script>
    const buildMenu = (tokens) => {
      const wrapper = document.getElementById('wrapper')
  
      tokens.forEach(tokenInfo => {
        const link = document.createElement('a');
        link.classList = 'nav-link';
        link.href = tokenInfo.path;
        link.innerHTML = tokenInfo.name
  
        const item = document.createElement('li');
        item.classList = 'nav-item'
        item.appendChild(link);
        wrapper.appendChild(item);
      })
    }
    window.onload = () => {
      fetch('../paths.json')
      .then(response => response.json())
      .then(buildMenu)
    }
  </script>  <div class=\\"container mt-5\\">
    <div class=\\"row\\">
      <div class=\\"color col-12 col-sm-6 col-md-4 col-lg-3\\">
        <div class=\\"color-sample\\" style=\\"background: #F091C9\\"></div>
        <p class=\\"color-name\\">colorPrimary: <code>#F091C9</code></p>
      </div>
    </div>
  </div>

  <script src=\\"https://code.jquery.com/jquery-3.5.1.slim.min.js\\"
    integrity=\\"sha384-DfXdz2htPH0lsSSs5nCTpuj/zy4C+OGpamoFVy38MVBnE+IbbVYUew+OrCXaRkfj\\"
    crossorigin=\\"anonymous\\"></script>
  <script src=\\"https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/js/bootstrap.min.js\\"
    integrity=\\"sha384-OgVRvuATP1z7JjHLkuOU7Xw704+h835Lr+6QL9UvYjZE3Ipu6Tp75j7Bh/kR0JKI\\"
    crossorigin=\\"anonymous\\"></script>
</body>

</html>"
`;
